src/common/auth/server.ts

```diff
@@ -12,24 +12,49 @@ You should have received a copy of the GNU General Public License along with @p0
 /** Implements a local auth server, which can receive auth tokens from an OIDC app */
 import { sleep } from "../../util";
 import express from "express";
+import { readFile } from "node:fs/promises";
 import http from "node:http";
-import { dirname, join } from "node:path";
+import { join, resolve } from "node:path";
+import { isSea, getAssetAsBlob } from "node:sea";
+import { Readable } from "node:stream";
 
-const ROOT_PATH = join(dirname(__filename), "..", "..");
+const ROOT_PATH = resolve(`${join(__dirname, "..", "..")}/public`);
+const LANDING_HTML_PATH = "redirect-landing.html";
+const FAVICON_PATH = "favicon.ico";
 
 /** A small amount of time is necessary prior to shutting down the redirect server to
  * properly render the redirect-landing page
  */
 const SERVER_SHUTDOWN_WAIT_MILLIS = 2e3;
 
+const pipeToResponse = (
+  bytes: Buffer,
+  res: express.Response,
+  contentType: string
+) => {
+  const stream = Readable.from(bytes);
+  res.setHeader("Content-Type", contentType);
+  res.setHeader("Content-Length", bytes.length);
+  stream.pipe(res);
+};
+
+const loadStaticAsset = async (path: string): Promise<Buffer> => {
+  if (isSea()) {
+    const blob = getAssetAsBlob(path);
+    return Buffer.from(await blob.arrayBuffer());
+  }
+  const filePath = join(ROOT_PATH, path);
+  const bytes = await readFile(filePath);
+  return bytes;
+};
+
 /** Waits for an OIDC authorization redirect using a locally mounted server */
 export const withRedirectServer = async <S, T, U>(
   start: (server: http.Server) => Promise<S>,
   complete: (value: S, token: T) => Promise<U>,
   options?: { port?: number }
 ) => {
   const app = express();
-  app.use(express.static(`${ROOT_PATH}/public`));
```

Comment:
Express.static assets aren't bundled into the executable, so a different solution is needed here. Otherwise the landing page after login won't work.